src/Compilers/VisualBasic/Portable/Parser/ParseExpression.vb

```diff
@@ -1627,7 +1627,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
 
             Debug.Assert(CurrentToken.Kind = SyntaxKind.FunctionKeyword OrElse
                          CurrentToken.Kind = SyntaxKind.SubKeyword,
-                         "ParseFunctionLambda called on wrong token.")
+                         NameOf(ParseFunctionOrSubLambdaHeader) & " called on wrong token.")
```

Comment:
@gafter The change is only present in debug builds, and as far can tell has no observable effects on the rest of the code.